data/json/items/classes/gun.json

```diff
@@ -4,22 +4,46 @@
     "type": "GUN",
     "name": "base pistol",
     "skill": "pistol",
-    "aim_speed": 4
+    "aim_speed": 4,
+    "valid_mod_locations": [ [ "accessories", 2 ], [ "barrel", 1 ], [ "bore", 1 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "magazine", 1 ], [ "muzzle", 1 ], [ "rail", 1 ], [ "sights", 1 ], [ "stock", 1 ], [ "underbarrel", 1 ] ]
   },
   {
     "abstract": "pistol_revolver",
     "copy-from": "pistol_base",
     "type": "GUN",
     "name": "revolver",
     "proportional": { "reload": 0.7 },
-    "extend": { "flags": [ "RELOAD_ONE", "RELOAD_EJECT", "NEVER_JAMS" ] }
+    "extend": { "flags": [ "RELOAD_ONE", "RELOAD_EJECT", "NEVER_JAMS" ] },
+    "//": "Revolvers exclude the muzzle location preventing installation of suppressors",
+    "valid_mod_locations": [ [ "accessories", 2 ], [ "barrel", 1 ], [ "bore", 1 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "sights", 1 ], [ "stock", 1 ] ]
   },
   {
     "abstract": "rifle_base",
     "type": "GUN",
     "name": "base rifle",
     "skill": "rifle",
-    "aim_speed": 8
+    "aim_speed": 8,
+    "valid_mod_locations": [ [ "accessories", 4 ], [ "barrel", 1 ], [ "bore", 1 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "magazine", 1 ], [ "muzzle", 1 ], [ "rail", 1 ], [ "sights", 1 ], [ "stock", 1 ], [ "underbarrel", 1 ] ]
```

Comment:
Moving mods to rifle base will require a bigger review. There are multiple rifles with gimmicks like being unable to use some/all slots.

It would be the best if you split that off into a separate PR. And then checked with some script which rifles change slots and how, because finding them all by hand could take a while.